internal/fleet/agent_policy/create.go

```diff
@@ -41,6 +46,41 @@ func (r *agentPolicyResource) Create(ctx context.Context, req resource.CreateReq
 		return
 	}
 
+	// The CREATE response may not include all fields (e.g., space_ids can be null in the response
+	// even when specified in the request). Read the policy back to get the complete state.
+	// Only do this if we got a valid ID from the create response.
+	if policy != nil && policy.Id != "" {
```

Comment:
IIUC this is the same as the core of the `Read` function. Are we able to extract this and re-use it here, in read, and presumably in `Update` as well?